src/QGONexum.py

```diff
@@ -256,50 +278,52 @@ def renameNodes(node_list, prefix):
             node.changeName(prefix + " " + str(idx))
         
     
-    def setNodeLists(self, cutoffs, prefixes):
+    def setNodeLists(self, cutoffs: list[str], prefixes: list[str]):
         
         for i, file in enumerate(self.files):
             
             self.nodes_lists.append(self.getNodes(file, self.prefixes[i], cutoffs[i]))
 
 
     def setNodeCoordDictionaries(self):
-
+        node_list: list[QAOANode]
         for node_list in self.nodes_lists:
 
             self.ncds.append(self.getNodeCoordDict(node_list))
 
     def setCoordsLists(self):
-
+        node_list: list[QAOANode]
         for node_list in self.nodes_lists:
             self.coords_lists.append(self.getCoordList(node_list))
 
-    def __init__(self, links, prefixes, cutoffs):
-
-        self.setFiles(links)
-        self.prefixes = prefixes
-        
-        
-        self.setNodeLists(cutoffs, prefixes)
-        self.setCoordsLists()
-        self.setNodeCoordDictionaries()
-
 
 class QGOGraph:
 
-    qgo_problem = None
+    qgo_problem: QGOProblem = None
 
-    edges_lists = []
-    edge_list_names = {}
+    edges_lists: list[tuple[list[tuple[QAOANode, QAOANode]]]] = []
+    edge_list_names: dict[tuple[QAOANode, QAOANode], str] = {}
 
-    edges_capacities = []
-    all_edge_capacities = {}
+    edges_capacities: list[dict[tuple[QAOANode, QAOANode], float]] = []
+    all_edge_capacities: dict[tuple[QAOANode, QAOANode], float] = {}
 
-    edges_labels = []
+    edges_labels: list[dict[tuple[QAOANode, QAOANode], str]] = []
     
     graph = nx.DiGraph()
+
+    def __init__(self, qgo_problem: QGOProblem, thresholds: list[int]):
+
+        self.qgo_problem = qgo_problem
+        self.thresholds = thresholds
+        
+        self.setEdgeListsCapacitiesLabels(thresholds=thresholds)
+
+        self.setAllEdgeCapacities()
+        
+        self.setEdgeListNames()
+        #print(self.edge_list_names)
```

Comment:
Moved the constructor to the top of method list.

Reply:
Thumbs up